### ISSUE TITLE
componentDidCatch: document & decompose info parameter

### ISSUE THREAD

**Issue by brandonbloom** [facebook/react#10461]
Some quick googling didn't show any explanation of the `info` parameter for `componentDidCatch(error, info)`. The only key in that object that I've seen so far is `componentStack`, which is useful, but is already composed. If I want to do some custom rendering of that information, I have to parse it.

I'd love to see `info.componentStack` become a more general `info.text` string and then have `componentStack` be broken down to an array of frames. Something like `[{component, createdBy}]`.

**Comment by bvaughn** [comment]
This is a new feature, to be introduced in 16 final. We'll add documentation for it soon but probably not until we release an RC.

I don't expect the 2nd parameter, (the one containing `componentStack`), to change in the short-term for what it's worth. We went with a named parameter for that because it gives us the option to add more information in a backwards-compatible way.

**Comment by brandonbloom** [comment]
I was not proposing to change it from an object with keys, I was proposing to add/rename keys.

```javascript
// example call from framework to my component
that.componentDidCatch(err, {
  text: 'Something Bag Happened\nat Foo (Bar) \n...',
  componentStack: [
    {component: 'Foo', createdBy: 'Bar'},
    ...
  ]
})

**Comment by bvaughn** [comment]
Understood! I was just offering an explanation for why we used a named-parameter for a single param. 😄 

`ReactFiberComponentTreeHook` has all of the info you mention so it wouldn't be hard to pass it along. (And I think this issue is great to keep around for one of us to follow-up on.)

**Comment by bvaughn** [comment]
Having given this more thought, I'm thinking the following change might be nice:
```js
type Source = {
  fileName: string,
  lineNumber: number,
};

type StackFrame = {
  name: string | null,
  ownerName: string | null,
  source: Source | null,
};

// Passed to the function injected via ReactFiberErrorLogger.injection.injectDialog:
// injectedFunction(capturedError: CapturedError);
type CapturedError = {
  componentName: ?string,
  componentStack: string,
  componentStackFrames: Array<StackFrame>,
  error: mixed,
  errorBoundary: ?Object,
  errorBoundaryFound: boolean,
  errorBoundaryName: string | null,
  willRetry: boolean,
};

// Passed to the componentDidCatch lifecycle hook:
// componentDidCatch(error: Error, info: ErrorInfo);
type ErrorInfo = {
  componentStack: string,
  componentStackFrames: Array<StackFrame>,
};
```

**Comment by gaearon** [comment]
Closing in favor of https://github.com/facebook/react/issues/10461 which has more activity.

**Comment by bvaughn** [comment]
This issue _is_ #10461 😄

**Comment by gaearon** [comment]
Oh man.

**Comment by gaearon** [comment]
https://github.com/facebook/react/issues/10922
